### ISSUE TITLE
PlayerController.m_AIState.closestWaypoint is invalid after loading a new game

### ISSUE THREAD

**Issue by markusobi** [REGoth-project/REGoth#143]
When the external `"npc_getnearestwp"` is called, it calls `npc.playerController->getClosestWaypoint();`.
When a new game is loaded via the menu the waypoint returned by `getClosestWaypoint()` can be invalid.
https://github.com/REGoth-project/REGoth/blob/c5cfadf30144b5ddef178ebb212e6842b1a22aba/src/logic/scriptExternals/Externals.cpp#L580

I have the impression that `m_AIState.closestWaypoint` is currently misused or has a misleading name. Especially this code doesn't seem to make much sense to me: https://github.com/REGoth-project/REGoth/blob/f4f5c2ec00afd46d9bde185a22ede9bd46e45e3b/src/logic/PlayerController.cpp#L225
Shouldn't `closestWaypoint` be always the closest waypoint? So either it should be reworked or renamed.
In other parts of REGoth someone uses this function and expects it to return the CLOSEST waypoint like in `npc_getnearestwp`.

Steps to reproduce a bug:

- get Mordrag quest from Thorus
- go to new camp with Mordrag until he reaches the last WP. Don't talk to him yet.
- save & load
- Now his "Here we are, take that ring to Lares" important talk won't show up, because his condition calls Npc_`GetNearestWP`

```
FUNC int  Org_826_Mordrag_AtNewcamp_Condition()
{
	if 	( Npc_KnowsInfo(hero,Org_826_Mordrag_GotoNewcamp) && Hlp_StrCmp(Npc_GetNearestWP(self),"OW_PATH_07_21") ) 
	{
		self.flags	= 0;
		return 1;
	};
};
```

**Comment by degenerated1123** [comment]
This code was written when the message-queue didn't exist and should be upgraded to the new system. There is even some old code in there regarding routines, which has been completely moved to the NPCs AI-State. The closest waypoint of an NPC should also be reflected in the `wp` filed in an NPCs script instance.

However, all of that doesn't matter, since it has to be redone for NPCs to walk around properly anyways.

**Comment by degenerated1123** [comment]
Fixed in PR #245
